core/src/controls.ts

```diff
@@ -69,7 +69,7 @@ export function controlSignalFromSpec(spec: ControlSignalSpec) {
  */
 export abstract class ControlSignal<A extends ControlSignalUserArgs> {
   readonly depth: number;  // Size of final output dimension.
-  abstract getTensors(numSteps: number, args: A): tf.Tensor2D;
+  abstract getTensors(numSteps: number, args?: A): tf.Tensor2D;
```

Comment:
This isn't right, because then getTensors in the subclasses has to be able to handle the case of missing args.

Reply:
Done.